core/src/epicli/data/common/ansible/playbooks/roles/kubernetes_master/tasks/cni-plugins/calico.yml

```diff
@@ -19,24 +19,19 @@
       when:
         - network_manager_conf_d_stat.stat.exists
 
-    - name: Reload NetworkManager service
-      block:
-        - name: Check if NetworkManager service is loaded
-          shell: >-
-            set -o pipefail &&
-            systemctl list-units --type=service --state=loaded | grep -q NetworkManager.service
-          args:
-            executable: /bin/bash
-          register: network_manager_service_status
-          changed_when: false
+- name: Reload NetworkManager service
+  block:
+    - name: Check if NetworkManager service is loaded
+      command: systemctl show NetworkManager.service --property=LoadState
```

Comment:
I'm not sure if the new command is an equivalent of the previous one.

Check this:
```
[ec2-user@ec2-35-181-55-88 ~]$ systemctl list-units --type=service --state=loaded | grep filebeat
filebeat.service                                      loaded active running Filebeat sends log files to Logstash or directly to Elasticsearch.
[ec2-user@ec2-35-181-55-88 ~]$ systemctl show filebeat --property=LoadState
LoadState=loaded
[ec2-user@ec2-35-181-55-88 ~]$ sudo systemctl stop filebeat
[ec2-user@ec2-35-181-55-88 ~]$ systemctl list-units --type=service --state=loaded | grep filebeat
[ec2-user@ec2-35-181-55-88 ~]$ echo $?
1
[ec2-user@ec2-35-181-55-88 ~]$ systemctl show filebeat --property=LoadState
LoadState=loaded
```
It looks like `--property=SubState` or `--property=ActiveState` could work:
```
[ec2-user@ec2-35-181-55-88 ~]$ systemctl show filebeat --property=ActiveState
ActiveState=inactive
[ec2-user@ec2-35-181-55-88 ~]$ systemctl show filebeat --property=SubState
SubState=dead
[ec2-user@ec2-35-181-55-88 ~]$ sudo systemctl start filebeat
[ec2-user@ec2-35-181-55-88 ~]$ systemctl show filebeat --property=ActiveState
ActiveState=active
[ec2-user@ec2-35-181-55-88 ~]$ systemctl show filebeat --property=SubState
SubState=running
[ec2-user@ec2-35-181-55-88 ~]$ systemctl show NetworkManager.service --property=SubState
SubState=running
[ec2-user@ec2-35-181-55-88 ~]$ systemctl show NetworkManager.service --property=ActiveState
ActiveState=active
[ec2-user@ec2-35-181-55-88 ~]$ systemctl show NetworkManagerTest.service --property=ActiveState
ActiveState=inactive
[ec2-user@ec2-35-181-55-88 ~]$ echo $?
0
systemctl show NetworkManagerTest.service --property=SubState
SubState=dead
```
I would consider keeping grep, since it's simple. However, `set -o pipefail` is unnecessary but the task should have condition `failed_when: network_manager_service_status.rc > 1` and maybe it could have a more accurate name, e.g. `Check if NetworkManager service needs to be reloaded`.

Reply:
Both checking loadstate but `systemctl list-units --type=service --state=loaded` tell us that service is loaded and active. We have to keep in mind that service can be loaded even if its inactive, but it cannot be reloaded. I will fix that :+1: